src/test/java/rx/internal/operators/SingleOnSubscribeUsingTest.java

```diff
@@ -0,0 +1,492 @@
+            }
+        };
+
+        Func1<Resource, Single<String>> observableFactory = new Func1<Resource, Single<String>>() {
+            @Override
+            public Single<String> call(Resource resource) {
+                return Single.just(resource.getTextFromWeb());
+            }
+        };
+
+        @SuppressWarnings("unchecked")
+        Observer<String> observer = mock(Observer.class);
+        Single<String> observable = Single.using(resourceFactory, observableFactory,
+                new DisposeAction(), disposeEagerly);
+        observable.subscribe(observer);
+        observable.subscribe(observer);
+
+        InOrder inOrder = inOrder(observer);
+
+        inOrder.verify(observer).onNext("Hello world!");
+        inOrder.verify(observer).onCompleted();
+
+        inOrder.verify(observer).onNext("Hello world!");
+        inOrder.verify(observer).onCompleted();
+        inOrder.verifyNoMoreInteractions();
+    }
+
+    @Test(expected = TestException.class)
+    public void withResourceFactoryError() {
+        performTestUsingWithResourceFactoryError(false);
+    }
+
+    @Test(expected = TestException.class)
+    public void withResourceFactoryErrorDisposeEagerly() {
+        performTestUsingWithResourceFactoryError(true);
+    }
+
+    private void performTestUsingWithResourceFactoryError(boolean disposeEagerly) {
+        Func0<Subscription> resourceFactory = new Func0<Subscription>() {
+            @Override
+            public Subscription call() {
+                throw new TestException();
+            }
+        };
+
+        Func1<Subscription, Single<Integer>> observableFactory = new Func1<Subscription, Single<Integer>>() {
+            @Override
+            public Single<Integer> call(Subscription subscription) {
+                return Single.just(1);
+            }
+        };
+
+        Single.using(resourceFactory, observableFactory, disposeSubscription)
+        .toBlocking().value();
+    }
+
+    @Test
+    public void withSingleFactoryError() {
+        performTestUsingWithSingleFactoryError(false);
+    }
+
+    @Test
+    public void withSingleFactoryErrorDisposeEagerly() {
+        performTestUsingWithSingleFactoryError(true);
+    }
+
+    private void performTestUsingWithSingleFactoryError(boolean disposeEagerly) {
+        final Action0 unsubscribe = mock(Action0.class);
+        Func0<Subscription> resourceFactory = new Func0<Subscription>() {
+            @Override
+            public Subscription call() {
+                return Subscriptions.create(unsubscribe);
+            }
+        };
+
+        Func1<Subscription, Single<Integer>> observableFactory = new Func1<Subscription, Single<Integer>>() {
+            @Override
+            public Single<Integer> call(Subscription subscription) {
+                throw new TestException();
+            }
+        };
+
+        try {
+            Single.using(resourceFactory, observableFactory, disposeSubscription)
+            .toBlocking().value();
```

Comment:
This file based on the OnSubscribeUsingTest. Besides, I like value() throwing instead of having some OnErrorNotImplementedException bubbling up.